### PR TITLE
Implement a pipeline step for recommending AVX2 builds of AICoE Tenso…

### PR DESCRIPTION
**Pull request by fridex**
…rFlow

### CODE REVIEW

File: thoth/adviser/steps/tensorflow_avx2.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        {
+            "type": "INFO",
+            "message": "AICoE TensorFlow builds are optimized for AVX2 instruction sets that your CPU supports",
+        }
+    ]
+
+    # A tuple (CPU_FAMILY, CPU_MODEL) of Intel processors supporting AVX2:
+    #   https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX2
+    #   https://en.wikichip.org/wiki/intel/cpuid
+    _AVX2_CPUS = frozenset(
```
Comment (fridex) [inline thoth/adviser/steps/tensorflow_avx2.py:56]: I'm not 100% sure with this listing - not sure if all the CPUs are captured correctly.

Comment (goern) [reply thoth/adviser/steps/tensorflow_avx2.py:56]: Please create an issue so that we don’t lose it, let’s check again later

Comment (goern) [reply thoth/adviser/steps/tensorflow_avx2.py:56]: @sub-mod

File: thoth/adviser/steps/tensorflow_avx2.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+
+class TensorFlowAVX2Step(Step):
+    """A step that recommends AICoE TensorFlow builds optimized for AVX2 enabled CPU processors."""
+
+    _SCORE_ADDITION = 0.2
+    _JUSTIFICATION_ADDITION = [
+        {
+            "type": "INFO",
+            "message": "AICoE TensorFlow builds are optimized for AVX2 instruction sets that your CPU supports",
```
Comment (fridex) [inline thoth/adviser/steps/tensorflow_avx2.py:49]: Is this message ok-ish? Maybe too personal with `your`? Do we want to have some convention on how messages should look like (besides schema used for justifications).

File: thoth/adviser/steps/tensorflow_avx2.py (hunk 1 above)
Comment (fridex) [reply thoth/adviser/steps/tensorflow_avx2.py:56]: The issue opened #1022

File: thoth/adviser/steps/tensorflow_avx2.py (hunk 2 above)
Comment (pacospace) [reply thoth/adviser/steps/tensorflow_avx2.py:49]: How the pipeline is triggered? Checking .thoth.yaml CPU family and model I guess, right?

"AICoE TensorFlow builds are optimized for AVX2 instruction sets supported in the CPU identified for the advise. "

Maybe something like this?

Not sure about a convention on the text itself in this case. (reqs: user-friendly, minimum details to give clear) understanding)

Comment (fridex) [reply thoth/adviser/steps/tensorflow_avx2.py:49]: Yes, based on the `.thoth.yaml` configuration.

Ack, will adjust the message :+1:

**Comment by goern** [comment]
@harshad16 @sub-mod can we create and publish AVX512 builds of TF? or Torch?

**Comment by harshad16** [comment]
> @harshad16 @sub-mod can we create and publish AVX512 builds of TF? or Torch?

The recent old releases of TensorFlow r2.x  wheels were with avx512 flags.
The pipeline needs some logic change and refactoring, will get on it.

**Comment by goern** [comment]
> > @harshad16 @sub-mod can we create and publish AVX512 builds of TF? or Torch?
> 
> The pipeline needs some logic change and refactoring, will get on it. 

Cloud you open an issue for that?

**Comment by harshad16** [comment]
> > > @harshad16 @sub-mod can we create and publish AVX512 builds of TF? or Torch?
> > 
> > 
> > The pipeline needs some logic change and refactoring, will get on it.
> 
> Cloud you open an issue for that?

sure, I will open an issue for that.

**Pull request merged**